features/0434-outofband/README.md

```diff
@@ -0,0 +1,417 @@
+
+## Tutorial
+
+### Key Concepts
+
+The out of band protocol is used when you don't know if you have a connection with another user. This could be because you are trying to establish a new connection with that agent, you have connections but don't know who the other party is, or if you want to have a connection-less interaction. Since there is no DIDComm connection to use for the messages of this protocol, the messages are plaintext and sent out of band, such as via a QR code. Since the delivery of out of band messages will often be via QR codes, this RFC also covers the use of QR codes.
+
+Two well known use cases for using the out of band protocol are:
+
+- A user on a web browser wants to execute a protocol (for example, to issue a verifiable credential) between the website operator's agent and a user's agent. To enable the execution of the protocol, the two agents must have a connection. An out-of-band message is used to trigger the creation of a new, or the reuse an existing, connection.
+- A website operator wants to request something from a user on their site where there is no (known) connection with the user.
+
+In both cases, there is only a single out of band protocol message sent. The message responding to the out of band message is a DIDComm message from an appropriate protocol.
+
+Note that the website-to-agent model is not the only such interaction enabled by the out of band protocol, and a QR code is not the only delivery mechanism for out of band messages. They are useful as examples of the purpose of the protocol.
+
+### Roles
+
+The out of band protocol has two roles: __sender__ and __receiver__.
+
+#### sender
+
+The agent that generates the out of band message and makes it available to the other party.
+
+#### receiver
+
+The agent that receives the out of band message and decides how to respond. There is no out of band protocol message with which the receiver will respond. Rather, if they respond, they will use a message from another protocol that the sender will understand.
+
+### States
+
+The state machines for the sender and receiver are a bit odd for the out of band protocol because of it being a single message protocol with a response from a different protocol. In the following state machine diagrams we generically describe the response message from the *receiver* as being a DIDComm message without associating the message with a specific protocol.
+
+The *sender* state machine is as follows:
+
+[![sender state machine google doc](state-machine-sender.png)](https://docs.google.com/spreadsheets/d/1o1szGCMx6FzrxfAKV76IEYmeKRJqvV3gd1r3Qfv1Q9o/edit#gid=1176419697)
+
+Note the "optional" reference under the second event in the `await-response` state. That is to indicate that an out of band message might be a single use message with a transition to done, or reusable message (received by many receivers) with a transition back to `await-response`.
```

Comment:
Do we need to worry at all about the possibility of a *resend*?

Reply:
I don't understand the context of the question. Could you please clarify?

Reply:
Never mind. Not important enough to worry about.